sequencer/src/block.rs

```diff
@@ -210,6 +231,23 @@ impl BlockHeader for Header {
         block_merkle_tree.push(parent.commit()).unwrap();
         let block_merkle_tree_root = block_merkle_tree.commitment();
 
+        // fetch receipts from the l1
+        let receipts = fetch_fee_receipts(parent);
+
+        let mut fee_merkle_tree = parent.fee_merkle_tree.clone();
+        for FeeReceipt { recipient, amount } in receipts {
+            // Get the balance in order to add amount. currently we
+            // are ignoring the proof. Should we verify it?
+            match fee_merkle_tree.universal_lookup(recipient) {
+                LookupResult::Ok(balance, _) => {
+                    fee_merkle_tree.update(recipient, balance + amount).unwrap()
+                }
+                // Handle `NotFound` and `NotInMemory` by initializing
+                // state. Should these be handled in different ways?
+                _ => fee_merkle_tree.update(recipient, amount).unwrap(),
+            };
```

Comment:
Not a big deal now, but this incurs an unnecessary tree traversal: one to lookup and one to update. Maybe we could add an API to jellyfish that takes a closure to update the existing value and a default value to insert if not existing? Then this update could all be done in a single pass. Not urgent. cc @mrain

Reply:
How about an interface like
```Rust
fn update(&mut self, f: F)
where
    F: FnOnce(Option<&E>) -> Option<E>;
```

`None` means that the key doesn't exist or we are removing it.